app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,34 @@
 class PostsController < ApplicationController
+  before_action :current_user, only: [:create]
   def index
     @user = User.find(params[:user_id])
-    @posts_list = @user.recent_posts
+    @posts_list = @user.recent_3_posts
   end
 
   def show
     @user = User.find(params[:user_id])
-    @post = Post.find(paraams[:id])
+    @post = Post.find(params[:id])
+  end
+
```

Comment:
- [ ] In order to create a post you need to add a new method here , it should be like this :point_down: 
```ruby 
def new
    @user = User.find(params[:user_id])
    @post = @user.posts.new
  end
```